### PR TITLE
Increase auto transitioned outputs amount to the minimum if it is lower

### LINKED ISSUE #1989: Does ISA need to handle chain outputs with less amount than minimum required amount?
Does ISA need to handle chain outputs with less amount than minimum required amount?
Currently it would just transition and then fail because the amount is not enough
`called `Result::unwrap()` on an `Err` value: Block(InsufficientStorageDepositAmount { amount: 1000, required: 87000 })`
(there are also other issues in this test, but it fails with this)
```Rust
#[test]
fn automatic_allot_account_mana_mana_requirement_twice() {
    let protocol_parameters = protocol_parameters();
    let account_id_1 = AccountId::from_str(ACCOUNT_ID_1).unwrap();

    let mut inputs = Vec::new();
    let mana_input_amount = 1_000;

    let account_output = AccountOutputBuilder::new_with_amount(2_000_000, account_id_1)
        .add_unlock_condition(AddressUnlockCondition::new(
            Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
        ))
        .with_mana(mana_input_amount)
        .finish_output()
        .unwrap();
    inputs.push(InputSigningData {
        output: account_output,
        output_metadata: rand_output_metadata_with_id(rand_output_id_with_slot_index(SLOT_INDEX)),
        chain: None,
    });

    let nft_output = NftOutputBuilder::new_with_amount(1_000, NftId::null())
        .add_unlock_condition(AddressUnlockCondition::new(
            Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
        ))
        .with_mana(30_000)
        .finish_output()
        .unwrap();
    inputs.push(InputSigningData {
        output: nft_output,
        output_metadata: rand_output_metadata_with_id(rand_output_id_with_slot_index(SLOT_INDEX)),
        chain: None,
    });

    let outputs = build_outputs([Basic {
        amount: 1_000_000,
        address: Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap(),
        native_token: None,
        sender: Some(Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap()),
        sdruc: None,
        timelock: None,
        expiration: None,
    }]);

    let selected = InputSelection::new(
        inputs.clone(),
        None,
        outputs.clone(),
        [Address::try_from_bech32(BECH32_ADDRESS_ED25519_0).unwrap()],
        SLOT_INDEX,
        SLOT_COMMITMENT_ID,
        protocol_parameters,
    )
    .with_auto_mana_allotment(account_id_1, 500)
    .select()
    .unwrap();

    assert!(unsorted_eq(&selected.inputs, &inputs));
    assert_eq!(selected.outputs.len(), 3);
    assert!(selected.outputs.contains(&outputs[0]));
    assert_eq!(selected.mana_allotments.len(), 1);
    let mana_cost = 230_000;
    assert_eq!(
        selected.mana_allotments[0],
        ManaAllotment::new(account_id_1, mana_cost).unwrap()
    );
    assert_eq!(selected.outputs[1].as_account().mana(), 0u64);
}

```

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

When ISA transitions outputs for you, it will now increase the amount of the created output to the minimum if it was lower.

### CODE REVIEW

File: sdk/tests/client/input_selection/account_outputs.rs [hunk 1]
```diff
@@ -2142,3 +2142,102 @@ fn implicit_account_transition() {
...
+        SLOT_INDEX,
+        SLOT_COMMITMENT_ID,
+        protocol_parameters.clone(),
+    )
+    .with_required_inputs([*inputs[0].output_id()])
+    .select()
+    .unwrap();
+
+    assert!(unsorted_eq(&selected.inputs_data, &inputs));
+    assert_eq!(selected.transaction.outputs().len(), 2);
```
Comment (Thoralf-M) [inline sdk/tests/client/input_selection/account_outputs.rs:2242]: Also check that the account output amount is now larger than small_amount?

Comment (DaughterOfMars) [reply sdk/tests/client/input_selection/account_outputs.rs:2242]: sure

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/types/block/output/basic.rs [hunk 2]
```diff
@@ -38,6 +38,11 @@ impl BasicOutputBuilder {
         Self::new(OutputBuilderAmount::Amount(amount))
     }
 
+    /// Creates an [`BasicOutputBuilder`] with a provided amount, unless it is below the minimum.
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/basic.rs]:
```suggestion
    /// Creates a [`BasicOutputBuilder`] with a provided amount, unless it is below the minimum.
```

File: sdk/src/types/block/output/foundry.rs [hunk 3]
```diff
@@ -97,6 +97,20 @@ impl FoundryOutputBuilder {
         Self::new(OutputBuilderAmount::Amount(amount), serial_number, token_scheme)
     }
 
+    /// Creates an [`FoundryOutputBuilder`] with a provided amount, unless it is below the minimum.
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/foundry.rs]:
```suggestion
    /// Creates a [`FoundryOutputBuilder`] with a provided amount, unless it is below the minimum.
```

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: sdk/src/types/block/output/delegation.rs [hunk 4]
```diff
@@ -41,13 +41,19 @@ impl DelegationId {
     }
 }
 
+#[derive(Copy, Clone)]
+pub enum DelegatedAmount {
```
Comment (thibault-martinez) [inline sdk/src/types/block/output/delegation.rs:46]: I'm not too convinced by this but I also think we should reverse amount/delegated amount and we have an issue for this so I'll ignore until we address it.

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**